File: services/python-worker/add_user_to_org.py
#!/usr/bin/env python

import sys

from common import neo4j_driver

if len(sys.argv) != 3:
  print("Usage: create-user USERNAME ORGANIZATION")
  exit()

username = sys.argv[1]
org = sys.argv[2]

print(f"Adding {username} to {org}")

session = neo4j_driver.session()
tx = session.begin_transaction()
tx.run('''
MATCH (o:Organization { name: $org }), (u:User { username: $username })
CREATE (u)-[:MEMBER]->(o)
''', username=username, org=org)
tx.commit()

File: server/tasks/add-user-to-org.ts
import { safeQuery } from '../neo4j/connection'

if (process.argv.length !== 4) {
  console.log('Usage: add-user-to-org USERNAME ORGANIZATION')
  process.exit(1)
}

const username = process.argv[2]
const org = process.argv[3]

console.log(`Adding ${username} to ${org}`)

safeQuery(`MATCH (o:Organization { name: $org }), (u:User { username: $username })
          CREATE (u)-[:MEMBER]->(o)`, { username, org })
  .then(() => process.exit(0))
  .catch(() => process.exit(1))
